src/config.ts

```diff
@@ -0,0 +1,242 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as jsonschema from 'jsonschema';
+import * as path from 'path';
+
+import {Browser} from './browser';
+import {parseHorizons} from './cli';
+import {CheckConfig} from './github';
+import {isUrl} from './specs';
+import {Horizons} from './stats';
+import {BenchmarkSpec, Measurement, PackageDependencyMap} from './types';
+import {fileKind} from './versions';
+
+/**
+ * Expected format of the top-level JSON config file. Note this interface is
+ * used to generate the JSON schema for validation.
+ */
+export interface ConfigFile {
+  root?: string;
```

Comment:
It's a bit of a pain, but if you duplicate the flag docs here, they will end up as documentation in the json schema, which we can leverage in editors.

Reply:
Done.